pkg/system/phase1_verifying.go

```diff
@@ -275,25 +275,28 @@ func (r *Reconciler) checkBucketLoggingPVC() error {
 			"'Guaranteed' BucketLogging requires a Persistent Volume Claim (PVC) with ReadWriteMany (RWX) access mode. Please specify the 'BucketLoggingPVC' to ensure guaranteed logging")
 	}
 
-	// Check if pvc exists in the cluster
-	BucketLoggingPVC := &corev1.PersistentVolumeClaim{
-		TypeMeta: metav1.TypeMeta{Kind: "PersistenVolumeClaim"},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      *r.NooBaa.Spec.BucketLogging.BucketLoggingPVC,
-			Namespace: r.Request.Namespace,
-		},
-	}
-	if !util.KubeCheck(BucketLoggingPVC) {
-		return util.NewPersistentError("InvalidBucketLoggingConfiguration",
-			"The specified 'BucketLoggingPVC' was not found")
-	}
+	if r.NooBaa.Spec.BucketLogging.BucketLoggingPVC != nil {
```

Comment:
Observation on line number 273 and on this condition + your comment on changes 
"Checking for bucket logging pvc only if BucketLoggingPVC is provided by the user. Currently, checks are failing if cephfs storage class is already present inside the cluster."

Let's say BucketLoggingPVC != nil so online 273 it will not check if cephfs sc is present or not.
Now, it will enter in this condition and will have he same impact. Right?

Is this what we want to check?

Reply:
Updated, Thanks.